src/ducks/export/services.js

```diff
@@ -0,0 +1,121 @@
+import { Q } from 'cozy-client'
+import { format } from '@fast-csv/format'
+import formatDate from 'date-fns/format'
+
+import { TRANSACTION_DOCTYPE } from 'doctypes'
+import categories from 'ducks/categories/tree'
+
+const DATE_FORMAT = 'YYYY-MM-DD'
+export const ROOT_DIR_ID = 'io.cozy.files.root-dir'
+
+const dateStr = date =>
+  date != null ? formatDate(date, DATE_FORMAT) : undefined
+
+/**
+ * Fetch all transactions and associated data.
+ *
+ * @param {CozyClient} client A CozyClient instance
+ *
+ * @return {Promise<Array<BankTransaction>>} Promise that resolves with an array of io.cozy.bank.operations documents
+ */
+export const fetchTransactionsToExport = async client => {
+  const transactions = await client.queryAll(
+    Q(TRANSACTION_DOCTYPE)
+      .include(['account', 'recurrence', 'tags'])
+      .limitBy(1000)
+  )
+  return client.hydrateDocuments(TRANSACTION_DOCTYPE, transactions)
+}
+
+/**
+ * Create a Transform Stream to transform input data into CSV lines.
+ *
+ * @return {TransformStream} The generated Transform Stream
+ */
+export const createFormatStream = () => {
+  return format({
+    delimiter: ';',
+    quoteColumns: true,
+    headers: [
+      'Date',
+      'Realisation date',
+      'Assigned date',
+      'Label',
+      'Original bank label',
+      'Category name',
+      'Amount',
+      'Currency',
+      'Type',
+      'Reimbursement status',
+      'Bank name',
+      'Account name',
+      'Account number',
+      'Account type',
+      'Recurrence name',
+      'Tag 1',
+      'Tag 2',
+      'Tag 3',
+      'Tag 4',
+      'Tag 5'
+    ]
+  })
+}
+
+/**
+ * Generator that transforms the given transactions into our own CSV format and yields the result line by line.
+ *
+ * @param {Array<BankTransaction>} transactions The list of transactions to transform
+ *
+ * @return {IterableIterator<Array<string|undefined>>}
+ */
+export const transactionsToCSV = function* (transactions) {
+  for (const tr of transactions) {
```

Comment:
nit: `  for (const transaction of transactions) {`

Reply:
Le but était d'avoir des lignes pas trop longues par la suite mais si ça complexifie trop la lecture alors partons sur le mot complet.